constraints.go

```diff
@@ -186,10 +186,47 @@ type constraintsDeserializationFunc func(map[string]interface{}) (*constraints,
 
 var constraintsDeserializationFuncs = map[int]constraintsDeserializationFunc{
 	1: importConstraintsV1,
+	2: importConstraintsV2,
 }
 
 func importConstraintsV1(source map[string]interface{}) (*constraints, error) {
-	fields := schema.Fields{
+	coerced, err := schema.FieldMap(constraintsSchemaV1(), constraintsDefaultsV1()).Coerce(source, nil)
+	if err != nil {
+		return nil, errors.Annotatef(err, "constraints v1 schema check failed")
+	}
+	valid := coerced.(map[string]interface{})
+
+	cores, err := constraintsValidCPUCores(valid)
+	if err != nil {
+		return nil, err
+	}
+
+	return validatedConstraints(1, valid, cores), nil
+}
+
+func importConstraintsV2(source map[string]interface{}) (*constraints, error) {
+	fields := constraintsSchemaV1()
+	fields["zones"] = schema.List(schema.String())
+
+	defaults := constraintsDefaultsV1()
+	defaults["zones"] = schema.Omit
+
```

Comment:
shouldn't these be layered behind a "constrainstSchemaV2" like V1 is being done?
and same for constraintsDefaultsV2 ?

I almost wonder if it should be:

func constraintsSchemaV1() ... {
  schema := constraintsSchemaV2()
  delete(schema["zones"])
}
following the general pattern of "make the newest version the easiest to read, and maintain backward compatible values".

Thoughts?